src/terraingenerator/main.cpp

```diff
@@ -73,28 +89,132 @@ static void showVersion()
 {
     const QStringList arguments = QCoreApplication::arguments();
+    bool inPriority = false;
+    bool inSource = false;
+    bool inCombine = false;
+    int argCount = 0;
+    QStringList combine;
+
 
     for (int i = 1; i < arguments.size(); ++i) {
         const QString &arg = arguments.at(i);
+
+        // Process source option.
+        if (inSource) {
+            if (arg.at(0) == QLatin1Char('-')) {
+                // Some other option is starting here.
+                if (argCount == 0) {
+                    qWarning() << "Missing arguments for --input option.";
+                }
+                inSource = false;
+            } else {
+                // Append the source file.
+                argCount++;
+                options.sources.append(arg);
+                //qDebug() << "adding source" << arg;
+            }
+
+            // If we should still process input list, skip the rest of the loop.
+            if (inSource) continue;
+        }
+
+        // Process priority option.
+        if (inPriority) {
+            if (arg.at(0) == QLatin1Char('-')) {
+                // Some other option is starting here.
+                if (argCount == 0) {
+                    qWarning() << "Missing arguments for --input option.";
+                }
+                inPriority = false;
+            } else {
+                // Append the terrain name.
+                argCount++;
+                options.terrainPriority.append(arg);
+               // qDebug() << "adding priority" << arg;
+            }
+
+            // If we should still process priority list, skip the rest of the loop.
+            if (inPriority) continue;
+
+        }
+
+        // Process combine option.
+        if (inCombine) {
+            if (arg.at(0) == QLatin1Char('-')) {
+                // Some other option is starting here.
+                if (argCount == 0) {
+                    qWarning() << "Missing arguments for --combine option.";
+                }
+                inCombine = false;
+                options.combineList.append(combine);
+                combine.clear();
+            } else {
+                // Append the terrain name.
+                argCount++;
+                combine.append(arg);
+                //qDebug() << "adding" << arg << "to combine";
+            }
+
+            // If we should still process combine list, skip the rest of the loop.
+            if (inCombine) continue;
+        }
+
+        // Process other options.
         if (arg == QLatin1String("--help") || arg == QLatin1String("-h")) {
             options.showHelp = true;
         } else if (arg == QLatin1String("--version")
                 || arg == QLatin1String("-v")) {
             options.showVersion = true;
+        } else if (arg == QLatin1String("-o")
+                || arg == QLatin1String("--output")) {
+            i++;
+            if (i >= arguments.size()) {
+                qWarning() << "Missing argument to" << arg << "option";
```

Comment:
You'll probably want to put a `return false;` here, to avoid crashing on the line below and because the tool should abort if the command line parameters are invalid (there were other warnings above for which I think it also makes sense to just return false immediately after printing the message).